toolchain/check/context.h

```diff
@@ -472,6 +472,15 @@ class Context {
     return scope_stack_.full_pattern_stack();
   }
 
+  // An ongoing impl lookup, used to ensure termination.
+  struct ImplLookupStackEntry {
+    SemIR::ConstantId type_const_id;
+    SemIR::ConstantId interface_const_id;
+  };
```

Comment:
Style: per [declaration order](https://google.github.io/styleguide/cppguide.html#Declaration_Order), I think structs belong at the top